scripts/generate_docs.sc

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env amm
+
+import java.time.Instant
+import ammonite.ops._
+import ammonite.ops.ImplicitWd._
+
+/**
+  * Docs generation process:
+  *
+  * 1. copy docs/docs and docs/_layouts/docs.html part from 1.3, 1.4 and 1.5 branches to the temp dirs
+  * 2. copy the latest release (1.5_ branch doc to the temp dir
+  * 3. generate the latest release docs
+  * 4. replace docs/docs with docs/docs from 1.3, 1.4, 1.5 and generate docs for the _sites/docs_version
+  */
+
+def makeTmpDir(): Path = {
+  val timestamp = Instant.now().toString.replace(':', '-')
+  val path = root/"tmp"/s"marathon-docs-build-$timestamp"
+  mkdir! path
+  path
+}
+
+val ignoredVersions = Set("1.6.0")
+
+/**
+  * getting 3 latest releases tags to build docs for
+  */
+val docsTargetVersions = %%('git, "tag", "-l", "--sort=version:refname").out.lines.filter(_.matches("""v[1-9]+\.\d+\.\d+""")).groupBy(v => v.drop(1).takeWhile {
+  var reachedMinorVersion = false
+  elem => elem match {
+    case c if c.isDigit => true
+    case c if c == '.' && !reachedMinorVersion =>
+      reachedMinorVersion = true
+      true
+    case _ => false
+  }
+}).mapValues{ tags => tags.last.drop(1)}.toList.sortBy(_._1).map(_._2).filterNot(v => ignoredVersions.contains(v)).reverse.take(3).reverse
+
+val latestReleaseVersion = docsTargetVersions.last
+
+val buildDir = makeTmpDir()
+
+val docsDir = pwd/up/"docs"
+
+def generateDocsForVersion(docsPath: Path, version: String, outputPath: String = "_site"): Unit = {
+  %('bundle, "exec", s"jekyll build --config _config.yml,_config.$version.yml -d $outputPath/$version/")(docsPath)
+}
+
+def tagNameForVersion(version: String) = {
+  s"tags/v$version"
+}
+
+
+// step 1: copy docs/docs to the respective dirs
+
+docsTargetVersions foreach { version =>
```

Comment:
What do you think about making these steps main methods as we've done in `ci/pipeline`?

Reply:
I think it's a good idea, but I'd rather do it in the next iteration. This script depends on other branches that aren't merged yet, and it needs to be improved after those branches will be merged.